scripts/cluster/agent.py

```diff
@@ -228,7 +228,10 @@ def is_valid(token_line, token_type=cluster_tokens_file):
 
     with open(token_type) as fp:
         for _, line in enumerate(fp):
-            if token == line.strip():
+            token_in_file = line.strip()
+            if "|" in line :
+                token_in_file = line.strip().split('|')[0]            
```

Comment:
Shouldn't we be checking the validity of the token before returning it? I think it would make the code much cleaner if we did this check here at the source.

Reply:
Yes we can remove all expired token just before checking for validity if that is fine.